### PR TITLE
Entrega Construindo um Modelo com Tensorflow - 🐶 🐱!

### PR DESCRIPTION
**Pull request by ryan-ribeiro**
Apresento duas soluções para o problema.

Percebi que o dataset de teste1 não estava com os labels, logo foi difícil encontrar uma lógica que resolvesse rapidamente o problema da acurácia sempre muito alta.

A solução v1 é uma "abordagem de aprendizado por reforço", no qual uma fração limitada do dataset de teste (100 imagens) vão passar pela predição do modelo, e o usuário irá avaliar se cada imagem está correta ou não. Ao final, é fornecido uma acurácia média do modelo. O modelo não está aprendendo com esse reforço do usuário. E ao tentar fazer o GPT fazer esse re-treino, a acurácia do modelo cai de 83% para 60%, mesmo sendo as mesmas imagens sendo mostradas na classificação logo em seguida.

A solução v2 utiliza três funções da tensorflow.keras para calcular a acurácia. Só encontrei essa solução posteriormente. Diferentemente da v1, existe um grande abismo entre a acurácia do treino do modelo (75%) e a acurácia do teste (52%). Não sei explicar o porque disso, mas acho que deve ter a ver com como a função está fazendo essa acurácia. Demora segundos para varrer os 12500 imagens e dizer uma acurácia do modelo. Muito estranho.

Pode me ajudar?

Sobre o exercício do livro, reenvio o projeto. Ele apresenta alguns erros no código principal, mas quando se compila no colab eles desaparecem. Existem alguns erros nos exercícios que eu não resolvi.

### CODE REVIEW

**Comment by ryan-ribeiro** [comment]
> Apresento duas soluções para o problema.
> 
> Percebi que o dataset de teste1 não estava com os labels, logo foi difícil encontrar uma lógica que resolvesse rapidamente o problema da acurácia sempre muito alta.
> 
> A solução v1 é uma "abordagem de aprendizado por reforço", no qual uma fração limitada do dataset de teste (100 imagens) vão passar pela predição do modelo, e o usuário irá avaliar se cada imagem está correta ou não. Ao final, é fornecido uma acurácia média do modelo. O modelo não está aprendendo com esse reforço do usuário.
> 
> A solução v2 utiliza **três funções da tensorflow.keras** para calcular a acurácia. Só encontrei essa solução posteriormente. Diferentemente da v1, existe um grande abismo entre a acurácia do treino do modelo e a acurácia do teste. Não sei explicar o porque disso, mas acho que deve ter a ver com como a função está fazendo essa acurácia. Demora segundos para varrer os 12500 imagens e dizer uma acurácia do modelo. Muito estranho.
> 
> Pode me ajudar?

**Comment by scadriano** [comment]
Relaxa. Tudo certo. Vamos para frente.
